packages/components/Menu/src/MenuItem/useMenuItem.ts

```diff
@@ -99,12 +114,23 @@ function getAccessibilityStateWorker(disabled: boolean, accessibilityState?: Acc
   return { disabled };
 }
 
-export const useHoverFocusEffect = (hovered: boolean, componentRef: React.MutableRefObject<any>) => {
+export const useHoverFocusEffect = (hovered: boolean, componentRef: React.MutableRefObject<any>, setIsMenuItemHovered: any) => {
```

Comment:
I feel like you could simplify this to:

```
export const useHoverFocusEffect = React.useLayoutEffect((hovered: boolean, componentRef: React.MutableRefObject<any>, setIsMenuItemHovered: any) => {
//etc.
```

instead of wrapping the useLayoutEffect hook in another function

Same with useRemoveHoverEffect